go/pkg/cosmos/websocket.go

```diff
@@ -10,23 +10,47 @@ import (
 	"github.com/cosmos/cosmos-sdk/simapp/params"
 	"github.com/pkg/errors"
 	"github.com/shapeshift/unchained/pkg/websocket"
-	tendermintjson "github.com/tendermint/tendermint/libs/json"
+	tmjson "github.com/tendermint/tendermint/libs/json"
 	coretypes "github.com/tendermint/tendermint/rpc/core/types"
 	tendermint "github.com/tendermint/tendermint/rpc/jsonrpc/client"
+	rpctypes "github.com/tendermint/tendermint/rpc/jsonrpc/types"
 	"github.com/tendermint/tendermint/types"
 )
 
 type TxHandlerFunc = func(tx types.EventDataTx, block *BlockResponse) (interface{}, []string, error)
+type UnmarshalResultEvent = func([]byte, *coretypes.ResultEvent) error
+
+type WebsocketHandler interface {
```

Comment:
websocket interface to suppport both tendermint websocket client and binance-sdk websocket clients